src/java/org/apache/cassandra/net/InboundSockets.java

```diff
@@ -202,7 +203,8 @@ private static void addBindings(InboundConnectionSettings template, ImmutableLis
         InboundConnectionSettings       settings = template.withDefaults();
         InboundConnectionSettings legacySettings = template.withLegacyDefaults();
 
-        if (settings.encryption.enable_legacy_ssl_storage_port)
+        if (settings.encryption.enable_legacy_ssl_storage_port &&
```

Comment:
I am wondering if it would be better to push this logic into `org.apache.cassandra.net.InboundConnectionSettings` rather than have here? or DatabaseDescriptor (only allow true if encryptions != none